src/voxel.cpp

```diff
@@ -40,12 +41,15 @@ VoxelManipulator::~VoxelManipulator()
 
 void VoxelManipulator::clear()
 {
-	// Reset area to volume=0
-	m_area = VoxelArea();
+	// Reset area to empty volume
+	VoxelArea old;
+	std::swap(m_area, old);
 	delete[] m_data;
 	m_data = nullptr;
 	delete[] m_flags;
 	m_flags = nullptr;
+
+	porting::TrackFreedMemory((sizeof(*m_data) + sizeof(*m_flags)) * old.getVolume());
```

Comment:
I don't think dereferencing a nullptr is a good idea here?

Reply:
No memory access happening, this is resolved at compile time. AFAIK it's also safe by the C++ standard.